modules/azure/loganalytics.go

```diff
@@ -0,0 +1,68 @@
+package azure
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/Azure/azure-sdk-for-go/services/preview/operationalinsights/mgmt/2020-03-01-preview/operationalinsights"
+)
+
+// LogAnalyticsWorkspaceExists indicates whether the operatonal insights workspaces exists; otherwise false or error
+func LogAnalyticsWorkspaceExists(workspaceName, resourceGroupName, subscriptionID string) bool {
+	ws, err := GetLogAnalyticsWorkspaceE(workspaceName, resourceGroupName, subscriptionID)
+	if err != nil {
+		return false
+	}
+
+	return (*ws.Name == workspaceName)
+}
+
+// GetLogAnalyticsWorkspaceSku return the log analytics workspace SKU as string as one of the following: Free, Standard, Premium, PerGB2018, CapacityReservation; otherwise empty string "".
+func GetLogAnalyticsWorkspaceSku(workspaceName, resourceGroupName, subscriptionID string) string {
+	ws, err := GetLogAnalyticsWorkspaceE(workspaceName, resourceGroupName, subscriptionID)
+	if err != nil {
+		return ""
+	}
+	return string(ws.Sku.Name)
+}
+
+// GetLogAnalyticsWorkspaceRetentionPeriodDays returns the log analytics workspace retention period in days; otherwise -1.
+func GetLogAnalyticsWorkspaceRetentionPeriodDays(workspaceName, resourceGroupName, subscriptionID string) int32 {
+	ws, err := GetLogAnalyticsWorkspaceE(workspaceName, resourceGroupName, subscriptionID)
+	if err != nil {
+		return -1
+	}
+	return *ws.RetentionInDays
+}
+
+// GetLogAnalyticsWorkspaceE gets an operational insights workspace if it exists in a subscription.
+// This function would fail the test if there is an error.
+func GetLogAnalyticsWorkspaceE(workspaceName, resoureGroupName, subscriptionID string) (*operationalinsights.Workspace, error) {
+	client, err := GetLogAnalyticsWorkspacesClientE(subscriptionID)
```

Comment:
use getTargetAzureSubscription to lookup subscriptionId

	subscriptionID, err := getTargetAzureSubscription(subscriptionID)
	if err != nil {
		return nil, err
	}

Reply:
the call stack will ultimately call getTargetAzureSubscription and not use the subscriptionID in the called method.